sdks/java/io/google-cloud-platform/build.gradle

```diff
@@ -116,8 +116,8 @@ dependencies {
   permitUnusedDeclared library.java.grpc_google_cloud_pubsublite_v1 // BEAM-11761
   implementation library.java.guava
   implementation library.java.http_client
-  provided library.java.hamcrest
-  permitUnusedDeclared library.java.hamcrest // BEAM-11761
+  implementation library.java.hamcrest
+  permitUnusedDeclared library.java.hamcrest // Needed for PipelineOptionRegistrar auto-service that registered TestPipelineOptions interfaces
```

Comment:
This is interesting. Somehow I though it was a direct dependency within the code. Is it actually something upstream that has hamcrest as a provided dependency still?

Reply:
Yes, it's java-core also have hamcrest as a provided dependency: https://github.com/apache/beam/blob/3aa78d21f70eb2e4236ac86281cf037896e6faed/sdks/java/core/build.gradle#L108

and the reflection access of TestPipeline sub-interfaces need hamcrest because it is a return type of a interface method: https://github.com/apache/beam/blob/3aa78d21f70eb2e4236ac86281cf037896e6faed/sdks/java/core/src/main/java/org/apache/beam/sdk/testing/TestPipelineOptions.java#L47